### DISCUSSION TITLE
A Neovim distribution for Neorg?

### DISCUSSION THREAD

**Discussion by v01dnu11** [nvim-neorg/neorg#810] [Ideas]
Hi everybody, Neorg is imho by far the most interesting project in the PKMS space at the moment.
Just as org-mode brought many people to Emacs, I see great potential for Neorg to bring new people to Neovim.
Unfortunately, it is somewhat complicated to install for a layperson, even if they have prior experience with Vim / Neovim.
I am therefore wondering whether it would be beneficial if we offered a plain init.lua that was only equipped with the necessary entries for a plugin manager and Neorg?

**Comment by pysan3** [comment]
If you want to go _Lazy_, I'd recommend using [LazyVim](https://github.com/LazyVim/LazyVim) and stick your neorg config inside the `~/.config/nvim/lua/plugins/` directory.

Create a file named `neorg-config.lua` (it could be anything) inside that directory, where the contents could be something like this.

https://github.com/pysan3/Norg-Tutorial/blob/main/norg_tutorial.md#kickstart-config

It is a tutorial of Neorg / Norg so if you are interested, I recommend read the whole page as well.

**Reply by v01dnu11** [reply]
Thank you for the suggestion, I hadn't thought about LazyVim. That is indeed a very fast way to get a complete and rather clean setup.
One could maybe also clone Vhyrro's  own neorg-dev repository or use Doom Nvim, which has Neorg enabled out of the box.

But those solutions come with their own baggage and might still be overwhelming for a newcomer.
Anyways, your input and especially your tutorial is much appreciated, thanks!

**Reply by pysan3** [reply]
> Doom Nvim, which has Neorg enabled out of the box.

Nice, didn't know that.

> might still be overwhelming for a newcomer

I could argue that nowadays, I believe it is easier to rather start from _nvim distributions_ (LazyVim, LunarVim etc) rather than from the ground up so giving instructions of [how to install via plugin managers](https://github.com/nvim-neorg/neorg#-installationquickstart) would be more suitable for that situation.

But I actually do get your point. I do see some people complaining about neorg for the difficulty of setup and lack of support of norg format on various platforms so I think this idea will decrease that anxiety a bit. There seems to be work going on to [create tutorials and stuffs](https://github.com/pysan3/Norg-Tutorial/commit/9b3c154b7df5709017288cb3d3ff9fbcc5eefdc0#commitcomment-111082225) so might well be added to the TODOs there.

Also, could you link to some `init.vim` equal file that explains how to setup org-mode over on emacs side for reference? With my zero-knowledge of emacs, the official org-mode site seems to only have instructions of [how to install it](https://orgmode.org/org.html#Installation) just like what is written in this README.

**Reply by v01dnu11** [reply]
You are right. There are several dependencies that should be fulfilled to get a "native" Neorg experience (at least a plugin manager, treesitter and a colorscheme).
Coincidentally (or not) d-r-a-b wrote a wonderful [page](https://github.com/nvim-neorg/neorg/wiki/Dependencies) for the wiki that explains how to install those together with Neorg.
So I think we are settled.

As a side note: While searching for a minimal distribution that could be used as an entry point [kickstart.nvim](https://github.com/nvim-lua/kickstart.nvim) came up. It is very clean, thoroughly commented and works beautifully with your template for a  Neorg setup.

> Also, could you link to some init.vim equal file that explains how to setup org-mode over on Emacs side for reference?

Afaik org-mode is already built into Emacs, no setup necessary. But I am no Emacs user anyways. I took it as an example because Neorg is inspired by org-mode.
